overwatch/processing/trending/objects/object.py

```diff
@@ -0,0 +1,111 @@
+import abc
+import logging
+import os
+import sys
+
+import ROOT
+
+import overwatch.processing.trending.constants as CON
+
+# https://stackoverflow.com/questions/35673474/using-abc-abcmeta-in-a-way-it-is-compatible-both-with-python-2-7-and-python-3-5/41622155#41622155
+if sys.version_info >= (3, 4):
+    ABC = abc.ABC
+else:
+    ABC = abc.ABCMeta('ABC', (object,), {'__slots__': ()})
+
+try:
+    from typing import *  # noqa
+    from overwatch.processing.processingClasses import histogramContainer  # noqa
+except ImportError:
+    pass
+
+logger = logging.getLogger(__name__)
+
+
+class TrendingObject(ABC):
+
+    def __init__(self, name, description, histogramNames, subsystemName, parameters):
+        # type: (str, str, list, str, dict) -> None
+        self.name = name
+        self.desc = description
+        self.histogramNames = histogramNames
+        self.subsystemName = subsystemName
+        self.parameters = parameters
+
+        self.currentEntry = 0
+        self.maxEntries = self.parameters.get(CON.ENTRIES, 50)
+        self.trendedValues = self.initStartValues()
+
+        self._histogram = None
+        self.drawOptions = 'AP'  # Ensure that the axis and points are drawn on the TGraph
+
+    def __str__(self):
+        return self.name
+
+    @abc.abstractmethod
+    def initStartValues(self):  # type: () -> Any
+        """Example:
+        return np.zeros((self.maxEntries, 2), dtype=np.float)
+        """
```

Comment:
When I changed this class to abstract, I convert method body to docstring. To be honest I didn't know until now that it is possible to have function with only docstring.
I will add 'pass', because I will be clean what function should do.

Reply:
Great, thanks!